kwasm-lemmas.md

```diff
@@ -390,21 +434,34 @@ module WRC20-LEMMAS
     imports KWASM-LEMMAS
 ```
 
-This conversion turns out to be helpful in this particular proof, but we don't want to apply it on all KWasm proofs.
+These conversions turns out to be helpful in this particular proof, but we don't want to apply it on all KWasm proofs.
 
 ```k
-    rule X /Int 256 => X >>Int 8
+    rule X /Int N => X >>Int 8  requires N ==Int 256 [simplification]
```

Comment:
Haskell backend shouldn't care if you have it as side condition or match directly.

Reply:
Maybe @ttuegel can confirm. On Java backend having `256` in the rule directly will generate an `==K` at unification time, but I'm pretty sure the Haskell backend will specialize this to `==Int` properly anyway.

Reply:
Some clarification would be great. I was under the impression simplification needed an exact match on the LHS to apply. So with the old rule, unless the denominator was the `Int` literal 256, this would not apply.

Reply:
Yes, `simplification` rules need an exact match. However, all rules must have their `requires` clause satisfied before they are applied, and if the backend can infer that `N ==Int 256`, then it likely will already have made that substitution into the term. So while the new rule is technically more general, I don't think it is likely to make a difference in practice. Do you have an example where it makes a difference?

Reply:
Ah, if `simplification` is with exact matches, it makes sense that the side-condition version will apply more often.